SAMstrict.tex

```diff
@@ -0,0 +1,398 @@
+
+\subsection{RNEXT}
+\samstrictrule{Invalid RNEXT}{If the template contains one segment RNEXT must be *}
+\samstrictrule{RNEXT not present in reference}{RNEXT must be equal to the value of one of the SQ SN values defined in the header.}
+\samstrictrule{Invalid RNEXT}{If the primary alignment of the next read in the template is mapped, RNEXT must not be {\tt *}}{\samrule}
+\samstrictrule{RNEXT not using =}{If the primary alignment of the next read in the template is mapped to the same reference sequence, RNEXT must be {\tt =}}{\samrule}
+\samstrictrule{Incorrect RNEXT}{If this read is unmapped or the primary alignment of the next read in the template is mapped to the a different reference sequence, RNEXT must match the RNAME of the next read.}
+
+\subsection{PNEXT}
+\samstrictrule{Invalid PNEXT}{If the template contains one segment PNEXT must be 0}
+\samstrictrule{Incorrect PNEXT}{If the primary alignment of the next read in the template is unmapped, PNEXT must be 0}
+\samstrictrule{Incorrect PNEXT}{If the primary alignment of the next read in the template is mapped, PNEXT must match the POS of that record.}
+
+\subsection{TLEN}
+\samstrictrule{TLEN out of range}{TLEN cannot be greater than 2147483647.}{\samrule}
+\samstrictrule{TLEN out of range}{TLEN cannot be less than -2147483648.}{\samrule}
+\samstrictrule{Invalid TLEN}{TLEN must be 0 if flag 0x1 is not set.}{\samrule}
+
+\subsection{SEQ}
+\samstrictrule{Inconsistent SEQ read lengths}{All alignments of a given segment must have consistent SEQ lengths. That is, for all non-* SEQ, SEQ + length of CIGAR hard clip must be equal. }
+\samstrictrule{Inconsistent SEQ sequences}{All alignments of a given segment with non-* SEQ must have consistent base calls. A base cannot be called an A in one record, but a T in another. Note that to determine the read base, both the 0x10 FLAG, and any hard clipping CIGAR operators need to be taken into account.}
+\samstrictrule{SEQ of secondary alignments specified.}{SEQ of secondary alignments (0x100 FLAG set) should be set to * to reduce the file size.}{\v15bestpractice}
+\samstrictrule{Invalid sequence base}{Unless SEQ is "*", SEQ read bases must be one of the following characters: acmgrsvtwyhkdbnACMGRSVTWYHKDBN}
+\samstrictrule{SEQ does not match reference when CIGAR indicates match.}{Unless SEQ is "*", read bases with CIGAR operator = must match the reference base. Bases are considered to match if overlap between the possible read and reference bases (based on their IUPAC codes) is non-zero.}
+\samstrictrule{SEQ matches reference when CIGAR indicates mismatch.}{Unless SEQ is "*", read bases with CIGAR operator X must not match the reference base. Bases are considered to match on if, when ignoring case, the reference and read bases are the same character  and the character is one of the following characters: acgtACGT.}
```

Comment:
period

Reply:
match ~on~ if,

also, I'm not sure I agree with your definition of match...you have a different definition of "match" in the line above (I like that one better...).

Reply:
This gets into the same murky waters as NM tag, specifically is N vs N a match?